### DISCUSSION TITLE
Generic prop type with non generic values

### DISCUSSION THREAD

**Discussion by genzzo** [react-hook-form/react-hook-form#9784] [Q&A]
I have a generic form component and a bunch of generic input components that accept a type T for their input names
However, I would like to assert that the name is both Path\<T> and another type at the same time (string, number, date...)

This is what my code looks like
```ts
type Props<T> = {
  name: Path<T>;
  required?: string | ValidationRule<boolean>;
};

const DateInput = <T extends FieldValues>({ name, required }: Props<T>) => {
  const { setValue, watch, register } = useFormContext<T>();

  const handleSetDate = (date: Date | null) => {
    if (!date) return console.log("date is null");
    setValue(name, date); // <---- Argument of type 'Date' is not assignable to parameter of type 'PathValue<T, Path<T>>'
  };

  return (
      <DatePicker
        {...register(name, { required })}
        selected={watch(name)}
        onChange={handleSetDate}
      />
  );
};
```

For this input, I would like to have name both **Path\<T>** and **Date** at the same time, but **Path\<T> & Date** doesn't work.
How can I can keep the generic solution without having to force a type cast (as PathValue<T, Path<T>>)?

**Comment by martin-wahlberg** [comment]
There is no way for the component to infer the type of the path and values of the form when you are not passing the entire form. Also since you want to update the value with the onchange makes me think that this is supposed to be a controlled component and you should probably use a controller instead of the register.

The FormElementComponent in my example in the link below may most likely solve your issue: 
[Examples in replies to this](https://github.com/react-hook-form/react-hook-form/discussions/8924 )
